app/assets/javascripts/shipit/deploy.js.coffee

```diff
@@ -1,7 +1,32 @@
-$(document).on 'click', '.deploy-abort .deploy-action', (event) ->
-  event.preventDefault()
-  $target = $(event.currentTarget)
-  return if $target.hasClass('pending')
-  $target.addClass('pending')
-  enable = -> $target.removeClass('pending')
-  $.post($target.attr('href')).done(-> setTimeout(enable, 3000))
+class AbortButton
+  SELECTOR = '[data-action="abort"]'
+
+  @handle: (event) =>
+    event.preventDefault()
+    $element = $(event.currentTarget).closest(SELECTOR)
```

Comment:
Is the `.closest` necessary given that you're binding to a click event on `[data-action="abort"]`?